contrib/babelfishpg_tds/src/backend/tds/tdsprotocol.c

```diff
@@ -77,6 +79,45 @@ static void ResetTDSConnection(void);
 static TDSRequest GetTDSRequest(bool *resetProtocol);
 static void ProcessTDSRequest(TDSRequest request);
 
+
+static void attention_listener_handler(SIGNAL_ARGS)
+{
+	int	save_errno;
+
+	if (TdsRequestCtrl->phase == TDS_REQUEST_PHASE_FETCH || TdsRequestCtrl->phase == TDS_REQUEST_PHASE_INIT || TdsRequestCtrl->phase == TDS_REQUEST_PHASE_FLUSH)
+		return ;
+
+	elog(LOG, "Attention signal received for process: %d", getpid());
```

Comment:
How chatty is this going to be? This feels like something that needs a GUC to control the logging level for this.